libp2p/netMessenger.go

```diff
@@ -147,16 +150,19 @@ func newNetworkMessenger(args ArgsNetworkMessenger, messageSigning messageSignin
 	if check.IfNil(args.PeersRatingHandler) {
 		return nil, fmt.Errorf("%w when creating a new network messenger", p2p.ErrNilPeersRatingHandler)
 	}
-
-	keyGen := crypto.NewIdentityGenerator()
-	p2pPrivateKey, err := keyGen.CreateP2PPrivateKey(args.P2pPrivateKeyBytes)
-	if err != nil {
-		return nil, err
+	if check.IfNil(args.P2pPrivateKey) {
+		return nil, fmt.Errorf("%w when creating a new network messenger", p2p.ErrNilP2pPrivateKey)
+	}
+	if check.IfNil(args.P2pSingleSigner) {
+		return nil, fmt.Errorf("%w when creating a new network messenger", p2p.ErrNilP2pSingleSigner)
+	}
+	if check.IfNil(args.P2pKeyGenerator) {
+		return nil, fmt.Errorf("%w when creating a new network messenger", p2p.ErrNilP2pSingleSigner)
```

Comment:
```suggestion
		return nil, fmt.Errorf("%w when creating a new network messenger", p2p.ErrNilP2pKeyGenerator)
```

perhaps extract a constant `baseErrorMessage := "when creating a new network messenger"` and reuse it in all `fmt.Errorf`s ?